### PR TITLE
[8.11] Adds knowledge base and Amazon Bedrock to AI Assistant docs

### LINKED ISSUE #4031: New Amazon Bedrock Connector
## Description 

In 8.11, the AI Assistant can use the new AWS Bedrock connector to connect to AWS Bedrock. 

Related:
- https://github.com/elastic/kibana/pull/166662

### ESS docs

- Will need to update the [AI Assistant docs](https://www.elastic.co/guide/en/security/8.10/security-assistant.html) wherever connectors are mentioned (e.g., Assistant requirements, how to set up the Assistant, etc.). 
- The AWS Bedrock connector docs are likely being handled by Steph or Lisa. (Haven't found a Kibana issue or PR tracking this work, but have reached out to Lisa.)

### Serverless docs
- The AWS Bedrock connector will be available in Serverless and is not going to be behind a feature flag.

## Additional notes from 8.11 feature sync

- Will be renamed (soon) to "Amazon Bedrock" connector
- We're only supporting the Anthropic models (Claude 2 and Claude Instant). An advantage of this model is it supports longer, more in-depth conversations (up to 100,000 tokens).
- Requirements on the AWS side:
   - Specific AWS role permissions (@jamesspi will provide details)
   - Enable the model in AWS
- We support any region that AWS supports (can link to their docs, either for this info or a general link elsewhere)

### LINKED ISSUE #4039: Knowledge Base for AI Assistant
## Description 

The Knowledge Base for AI Assistant helps users get answers on topics that aren't included in third-party AI models. The big reason for this is the brand-new ES|QL language (which LLMs don't know about yet), but can eventually include other targeted topics.

Knowledge Base is configured in a new tab on the settings modal for AI Assistant.

AFTER YOU SET UP KNOWLEDGE BASE, THERE IS A NEW QUICK PROMPT. YOU SHOULD USE IT FOR ESQL QUERIES. ALSO JAMES IS GOING TO PROVIDE EXAMPLES OF DIALOGUE WITH THE KNOWLEDGE BASE. 

## Requirements

@jamesspi can provide more specifics on these steps:

- Enable ELSER (in Machine Learning area of Kibana)
- Initialize the Knowledge Base
- Initialize the ES|QL Knowledge Base documents

All three of these are required for the Knowledge Base. Users can disable the entire feature if it's producing unexpected results, and AI Assistant will resume default functionality.

NOTE: NOT AVAILABLE IN SERVERLESS

### PR DESCRIPTION
**Pull request by benironside**
Fixes #4031 and fixes #4039 by updating the AI Assistant docs to include the Amazon Bedrock connector and the Knowledge base.

Preview: [Elastic AI Assistant](https://security-docs_4149.docs-preview.app.elstc.co/guide/en/security/master/security-assistant.html)

### CODE REVIEW

File: docs/assistant/security-assistant.asciidoc [hunk 1]
```diff
@@ -52,15 +52,39 @@ NOTE: Elastic can automatically anonymize event data that you provide to AI Assi
...
 == Set up AI Assistant
 
-You must complete these steps before you can use AI Assistant:
+You must create a generative AI connector before you can use AI Assistant. 
 
-. Create an API key with your AI provider to authenticate requests from AI Assistant. You'll use this in the next step. Refer to the provider's documentation for generating API keys:
-+
-* https://platform.openai.com/docs/api-reference[OpenAI]
-* https://learn.microsoft.com/en-us/azure/cognitive-services/openai/reference[Azure OpenAI Service]
+. Open AI Assistant *Cmd + ;* (or *Ctrl + ;* on Windows), and click **Connector** *->* *Add new Connector**
```
Comment (joepeeples) [inline docs/assistant/security-assistant.asciidoc:57]: Syntax fix:
```suggestion
. Open AI Assistant *Cmd + ;* (or *Ctrl + ;* on Windows), and click **Connector** -> **Add new Connector**.
```

File: docs/assistant/security-assistant.asciidoc [hunk 2]
```diff
@@ -52,15 +52,39 @@ NOTE: Elastic can automatically anonymize event data that you provide to AI Assi
...
+. Select either *Amazon Bedrock* or *OpenAI*. 
+. Enter the authentication details required for your chosen connector type, then click *Save*.
+
+For OpenAI and Azure OpenAI Service, you need to provide an API key. For Amazon Bedrock, you need to provide an access key and secret for an IAM user with at least the following permissions:
+
+.Expand permissions JSON
```
Comment (joepeeples) [inline docs/assistant/security-assistant.asciidoc:63]: IMO this example is short enough that it doesn't need to be collapsible, but it's also fine like this. If you keep it collapsible, maybe instruct the reader a bit more directly:
```suggestion
.Click to expand permissions JSON
```

File: docs/assistant/security-assistant.asciidoc [hunk 3]
```diff
@@ -52,15 +52,39 @@ NOTE: Elastic can automatically anonymize event data that you provide to AI Assi
...
+                "bedrock:InvokeModelWithResponseStream"
+            ],
+            "Resource": "*"
+        }
+    ]
+}
+```
+====
+
+For Amazon Bedrock, only Anthropic models are supported: `Claude`, and `Claude Instant`. You need to https://docs.aws.amazon.com/bedrock/latest/userguide/model-access.html#manage-model-access[enable them in AWS] before setting up an Amazon Bedrock connector. You can configure an Amazon Bedrock connector to use any https://docs.aws.amazon.com/general/latest/gr/bedrock.html[AWS region where Anthropic models are supported] by editing the *URL* field under *Connector settings*, for example by changing `us-west-2` to `eu-central-1`.
```
Comment (joepeeples) [inline docs/assistant/security-assistant.asciidoc:84]: The model names prob don't need formatting: monospace is typically for things the user types, and bold for clicking/selecting UI elements, and it doesn't look like the user is doing either here.

Also, if the user has to enable the models _before_ they set up a connector, we should tell them that _before_ the connector instructions.
```suggestion
For Amazon Bedrock, only Anthropic models are supported: Claude, and Claude Instant. You need to https://docs.aws.amazon.com/bedrock/latest/userguide/model-access.html#manage-model-access[enable them in AWS] before setting up an Amazon Bedrock connector. You can configure an Amazon Bedrock connector to use any https://docs.aws.amazon.com/general/latest/gr/bedrock.html[AWS region where Anthropic models are supported] by editing the *URL* field under *Connector settings*, for example by changing `us-west-2` to `eu-central-1`.
```

File: docs/assistant/security-assistant.asciidoc [hunk 4]
```diff
@@ -139,5 +163,19 @@ The fields on this list are among those most likely to provide relevant context
 image::images/add-alert-context.gif[A video that shows an alert being added as context to an AI Assistant chat message]
 +
 When you include a particular event as context, you can use a similar interface to adjust anonymization behavior. Be sure the anonymization behavior meets your specifications before sending a message with the event attached.
-
++
 The *Show anonymized* toggle controls whether you see the obfuscated or plaintext versions of the fields you sent to AI Assistant. It doesn't control what gets obfuscated — that's determined by the anonymization settings. It also doesn't affect how event fields appear _before_ being sent to AI Assistant. Instead, it controls how fields that were already sent and obfuscated appear to you.
+
+* **Knowledge base:** Use retrieval augmented generation to provide specialized knowledge of the Elastic Search Query Language ({esql}) to AI Assistant. For example, with knowledge base active, you can ask AI Assistant to help you write an {esql} query for a particular use-case, or ask it to answer general questions about {esql} syntax and usage. Without knowledge base enabled, AI Assistant will not be able to answer questions about {esql}.
```
Comment (joepeeples) [inline docs/assistant/security-assistant.asciidoc:169]:
```suggestion
* **Knowledge base:** Use retrieval-augmented generation to provide specialized knowledge of the Elastic Search Query Language ({esql}) to AI Assistant. For example, with the knowledge base active, you can ask AI Assistant to help you write an {esql} query for a particular use case, or ask it to answer general questions about {esql} syntax and usage. Without the knowledge base enabled, AI Assistant will not be able to answer questions about {esql}.
```

File: docs/assistant/security-assistant.asciidoc [hunk 5]
```diff
@@ -139,5 +163,19 @@ The fields on this list are among those most likely to provide relevant context
...
++
+beta::[]
++
+To enable Knowledge base:
```
Comment (joepeeples) [inline docs/assistant/security-assistant.asciidoc:173]:
```suggestion
To enable the knowledge base:
```

File: docs/assistant/security-assistant.asciidoc [hunk 6]
```diff
@@ -139,5 +163,19 @@ The fields on this list are among those most likely to provide relevant context
...
++
+- Enable the Elastic Learned Sparse EncodeR (ELSER). This model provides additional context to the third-party LLM. To learn more, refer to https://www.elastic.co/guide/en/machine-learning/current/ml-nlp-elser.html#download-deploy-elser[Configure ELSER]
```
Comment (joepeeples) [inline docs/assistant/security-assistant.asciidoc:175]:
```suggestion
- Enable the Elastic Learned Sparse EncodeR (ELSER). This model provides additional context to the third-party LLM. To learn more, refer to https://www.elastic.co/guide/en/machine-learning/current/ml-nlp-elser.html#download-deploy-elser[Configure ELSER].
```

File: docs/assistant/security-assistant.asciidoc [hunk 7]
```diff
@@ -139,5 +163,19 @@ The fields on this list are among those most likely to provide relevant context
...
+- Initialize the knowledge base by clicking *Initialize*.
+- Initialize the {ESQL} knowledge base documents by clicking *Initialize*.
+- Move the *Knowledge Base* slider to the `on` position.
```
Comment (joepeeples) [inline docs/assistant/security-assistant.asciidoc:178]: I think this is just a toggle that the user clicks, not a slider that they drags back and forth.
```suggestion
- Turn on the *Knowledge Base* option.
```

File: docs/assistant/security-assistant.asciidoc [hunk 8]
```diff
@@ -139,5 +163,19 @@ The fields on this list are among those most likely to provide relevant context
...
+- Click *Save*. Knowledge base is now active.
++
+When Knowledge base is active, a quick prompt for {esql} queries becomes available. It provides a good starting point for your {esql} conversations and questions.
```
Comment (joepeeples) [inline docs/assistant/security-assistant.asciidoc:181]:
```suggestion
- Click *Save*. The knowledge base is now active.
+
When the knowledge base is active, a quick prompt for {esql} queries becomes available. It provides a good starting point for your {esql} conversations and questions.
```

**Review by joepeeples — COMMENTED** [review]
Good content, just a few suggested edits. Thanks!

We need to decide (with @jamesspi's help) at least 2 terminology details for the knowledge base, and then standardize throughout:
- Any capitalization?
- Is it "the knowledge base" or just "knowledge base"?

IMO no capitalization and including "the" makes sense, so my suggestions do that, but we could revise if needed.

File: docs/assistant/security-assistant.asciidoc [hunk 9]
```diff
@@ -139,5 +163,19 @@ The fields on this list are among those most likely to provide relevant context
...
++
 The *Show anonymized* toggle controls whether you see the obfuscated or plaintext versions of the fields you sent to AI Assistant. It doesn't control what gets obfuscated — that's determined by the anonymization settings. It also doesn't affect how event fields appear _before_ being sent to AI Assistant. Instead, it controls how fields that were already sent and obfuscated appear to you.
+
+* **Knowledge base:** Use retrieval-augmented generation to provide specialized knowledge of the Elastic Search Query Language ({esql}) to AI Assistant. For example, with the knowledge base active, you can ask AI Assistant to help you write an {esql} query for a particular use case, or ask it to answer general questions about {esql} syntax and usage. Without the knowledge base enabled, AI Assistant will not be able to answer questions about {esql}.
++
+beta::[]
++
+To enable the knowledge base:
++
+- Enable the Elastic Learned Sparse EncodeR (ELSER). This model provides additional context to the third-party LLM. To learn more, refer to https://www.elastic.co/guide/en/machine-learning/current/ml-nlp-elser.html#download-deploy-elser[Configure ELSER].
```
Comment (natasha-moore-elastic) [inline docs/assistant/security-assistant.asciidoc:175]: Should we use the {ml-docs} attribute for the `https://www.elastic.co/guide/en/machine-learning` reference here?

Comment (natasha-moore-elastic) [inline docs/assistant/security-assistant.asciidoc:175]: Are these sequential steps, ie. should they be numbered?

**Review by natasha-moore-elastic — COMMENTED** [review]
Just a couple of questions, otherwise LGTM!

File: docs/assistant/security-assistant.asciidoc [hunk 10]
```diff
@@ -52,15 +52,39 @@ NOTE: Elastic can automatically anonymize event data that you provide to AI Assi
...
+                "bedrock:InvokeModelWithResponseStream"
+            ],
+            "Resource": "*"
+        }
+    ]
+}
+```
+====
+
+For Amazon Bedrock, only Anthropic models are supported: Claude, and Claude Instant. You need to https://docs.aws.amazon.com/bedrock/latest/userguide/model-access.html#manage-model-access[enable them in AWS] before setting up an Amazon Bedrock connector. You can configure an Amazon Bedrock connector to use any https://docs.aws.amazon.com/general/latest/gr/bedrock.html[AWS region where Anthropic models are supported] by editing the *URL* field under *Connector settings*, for example by changing `us-west-2` to `eu-central-1`.
```
Comment (joepeeples) [inline docs/assistant/security-assistant.asciidoc:84]:
```suggestion
For Amazon Bedrock, only Anthropic models are supported: Claude and Claude Instant. You need to https://docs.aws.amazon.com/bedrock/latest/userguide/model-access.html#manage-model-access[enable them in AWS] before setting up an Amazon Bedrock connector. You can configure an Amazon Bedrock connector to use any https://docs.aws.amazon.com/general/latest/gr/bedrock.html[AWS region where Anthropic models are supported] by editing the *URL* field under *Connector settings*, for example by changing `us-west-2` to `eu-central-1`.
```

**Review by jamesspi — APPROVED** [review]

File: docs/assistant/security-assistant.asciidoc (hunk 9 above)
Comment (spong) [reply docs/assistant/security-assistant.asciidoc:175]: Small note here -- we try to do this all in one step for the user if we can. So users don't need to click `initialize` for the KB or ESQL docs unless an error has occurred or they previously deleted the ESQL docs or KB index/pipeline. Once ELSER is enabled, they will only need to flip the main KB switch and then click save:


Kapture 2023-11-06 at 14 45 36

**Pull request merged**